### ISSUE TITLE
look like the _process_message_thread won't stop.

### ISSUE THREAD

**Issue by imfht** [jborean93/smbprotocol#32]
https://github.com/jborean93/smbprotocol/blob/4f7b81bebfa2f681d3ea645d4728c32b35bf5a2c/smbprotocol/connection.py#L875

if I run this code : 
```
connection = Connection(uuid.uuid4(), ip, 445)
while True:
    connection.connect(Dialects.SMB_3_1_1, timeout=15)
    connection.disconnect()
```
Loooots of new thread will be created.

**Comment by jborean93** [comment]
Thanks for bringing this up, a thread should definitely be linked to the connection and stopped once the connection is disconnected.

**Comment by jborean93** [comment]
Sorry for the delay but I had another look into this and the threads for each connection will be stopped when you run `.disconnect()`. In the `diconnect()` function it calls `self.transport.close()` https://github.com/jborean93/smbprotocol/blob/4dff793d76efff9d8f8a1b755d5c9e5ff0236d16/smbprotocol/connection.py#L804

Part of closing the transport is to close the socket by sending the shutdown signal, waiting for that thread to finish then finally closing the socket https://github.com/jborean93/smbprotocol/blob/4dff793d76efff9d8f8a1b755d5c9e5ff0236d16/smbprotocol/transport.py#L91-L93

When the actual socket is shutdown by either the client or the server, it is set to always add `None` to the connection thread queue https://github.com/jborean93/smbprotocol/blob/4dff793d76efff9d8f8a1b755d5c9e5ff0236d16/smbprotocol/transport.py#L134-L136

Finally once the `recv_queue` receives `None` (the socket has been closed) it exits the process which subsequently stops the thread https://github.com/jborean93/smbprotocol/blob/4dff793d76efff9d8f8a1b755d5c9e5ff0236d16/smbprotocol/connection.py#L1110-L1114

If you are finding this is not the case on your setup I'm happy to have a further look but it does seem like the thread will be cleaned up when you disconnect from the connection. Even when running a test I found that the only thread still running after calling `.disconnect()` is the main thread.
